packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -163,7 +163,7 @@ function collectExternals(
       }
 
       logger.warn(
-        `${id} is incorrectly packaged. Please contact the package author to fix.`
+        `${id} is not written in CJS, but also doesn't have "type": "module" or an .mjs extension for the entry point. Please contact the package author to fix.`
```

Comment:
suggest to make it more clear that `.mjs` extension and `"type": "module"` are for esm only
```suggestion
        `${id} have not a exported CJS build, but In case of esm only package, it doesn't have "type": "module" or an .mjs extension for the entry point. Please contact the package author to fix.`
```

Reply:
I think this would make the message less clear because it wouldn't align directly to the logic anymore. The logic just checks if it's CJS, type:module, or .mjs. If you're trying to use something that's not CJS or ESM that's probably an issue by itself

Reply:
I don't understand your comment, I suggested the warning to be more explicit about `.mjs` extension and `"type": "module"` are for ESM only.
it looked like @wildwestrom got confused in https://github.com/rasendubi/uniorg/issues/24 about this warning and thought it was only related a CJS build missing

Reply:
For https://github.com/rasendubi/uniorg/issues/24, it would be fixed by #6053. But for this warning, it seems to be for packages that looks like it's exporting CJS, but are exporting ESM code.

Maybe we can rephrase it as: `${id} is expected to export CJS code from the entrypoint: ${esmEntry}, but it is not. Please contact the package author to fix this by using "type": "module" or the .mjs extension.`

Reply:
I've updated the warning. Hopefully it makes more sense to you now